### PR TITLE
[test-suite] make green on Android

### PR DESCRIPTION
**Pull request by esamelson**
# Why

Lots of tests have been flaky/broken on Android for a while -- in preparation for other people to work on the release process, let's get everything as green as we can for now! Having flaky (or falsely negative) tests is costly for whoever is running the test and reduces the value of the test suite as a whole.

# How

- Audio: one test was broken because the link didn't exist anymore. I've changed it to a link that exists now but it might be good to find something more permanent for this.
- HTMLElements: `BR` is a text element, and it was in the wrong section (failing with `Views nested within a <Text> element must have a width and height`). Fixed by putting it in the correct test section.
- Linking: Added a missing timeout -- without this the test was consistently failing as there wasn't enough time for the callback to fire.
- Speech: These tests are consistently flaky, but increasing the timeout for Android seems to make them pass.
- ImagePicker: Added `await`ing on the alert, otherwise the instructions are not useful because the user doesn't have a chance to see them!
- Location: Some of the "simultaneous calls" tests are pretty inconsistent, so I disabled them for now.
- Camera: All of the `takePictureAsync` tests are flaky on Android. Tried adding longer timeouts, individual `_instance` variables per test but couldn't find anything that made a difference, so just disabled them for now.
- The FBNativeAd tests have failed consistently on Android for at least the last year. From my investigation it looks like they rely on the `NativeAdView.onAdLoaded` prop which actually only exists in the iOS implementation, so I disabled them all for now. Followup issue: https://github.com/expo/expo/issues/8643

- MediaLibrary: These tests are quite flaky. It turns out that sometimes the logic in the `beforeAll` function fails, which causes all of the tests to be red. Adding some retry logic reduces the failure rate here, but even then, sometimes individual tests fail seemingly at random. I'm debating just disabling all of these tests on Android for now, because they're not useful if they're flaky. But it might be good to do a deeper dive into why they're flaky -- I think there might be a race condition somewhere inside of the `AsyncTask`s in this module's implementation.

# Test Plan

All tests are green now! (Except for MediaLibrary... sometimes)

### CODE REVIEW

File: apps/test-suite/tests/HTML.js [hunk 1]
```diff
@@ -77,7 +78,6 @@ const viewElements = {
   Main,
   Section,
   Nav,
-  BR,
```
Comment (sjchmiela) [inline apps/test-suite/tests/HTML.js:80]: We may want to look again into https://github.com/expo/expo/pull/7395 which (I hope) provides even more test coverage to `BR`. Would you take another look at it, please, @EvanBacon? 🙂

File: apps/test-suite/tests/ImagePicker.js [hunk 2]
```diff
@@ -1,13 +1,19 @@
...
 import * as Permissions from 'expo-permissions';
-import { Platform } from 'react-native';
+import { Alert, Platform } from 'react-native';
 
 import * as TestUtils from '../TestUtils';
 import { isDeviceFarm } from '../utils/Environment';
 
 export const name = 'ImagePicker';
 
+async function alertWithInstructions(instructions) {
```
Comment (tsapeta) [inline apps/test-suite/tests/ImagePicker.js:11]: Might be worth moving to TestUtils or somewhere else so we can reuse it in other tests. I think we should show alerts also on some other interactive tests.

File: apps/test-suite/tests/ImagePicker.js [hunk 3]
```diff
@@ -55,14 +61,14 @@ export async function test({ it, xit, beforeAll, expect, jasmine, xdescribe, des
     describe('launchCameraAsync', () => {
       if (Constants.isDevice) {
         it('launches the camera', async () => {
-          alert('Please take a picture for this test to pass.');
+          await alertWithInstructions('Please take a picture for this test to pass.');
```
Comment (tsapeta) [inline apps/test-suite/tests/ImagePicker.js:64]: Sorry for that, I've added `non-await` alert during iOS QA where it worked and haven't tested it on Android 🤦‍♂️

File: apps/test-suite/tests/MediaLibrary.js [hunk 4]
```diff
@@ -94,18 +96,46 @@ export async function test(t) {
...
+
+    async function cleanupAsync() {
+      await MediaLibrary.deleteAssetsAsync(testAssets);
+      await MediaLibrary.deleteAlbumsAsync(album);
+    }
+
+    t.beforeAll(async () => {
+      // NOTE(2020-06-03): The `initializeAsync` function is flaky on Android; often the
+      // `addAssetsToAlbumAsync` method call inside of `createAlbum` will fail with the error
+      // "Could not get all of the requested assets". Usually retrying a few times works, so we do
```
Comment (tsapeta) [inline apps/test-suite/tests/MediaLibrary.js:119]: Just created an issue for this: https://github.com/expo/expo/issues/8653
I've assigned myself as I'd like to revisit some modules that I'm an owner of.

**Review by tsapeta — APPROVED** [review]

**Pull request merged**